src/docs/development/platform-integration/add-to-app-android/add-flutter-screen.md

```diff
@@ -0,0 +1,300 @@
+---
+title: Adding a Flutter screen to an Android app
+short-title: Add a Flutter screen
+description: Learn how to add a single Flutter screen to your existing Android app.
+---
+
+This guide describes how to add a single Flutter screen to an existing Android
+app. A Flutter screen can be added as a normal, opaque screen, or as a
+see-through, translucent screen. Both options are described in this guide.
+
+## Add a normal Flutter screen
+
+{% asset
+development/platform-integration/add-to-app-android/add-flutter-screen/add-single-flutter-screen_header.png
+class="mw-100" alt="Add Flutter Screen Header" %}
+
+### Step 1: Add Gradle dependencies
+
+TODO: update this with gradle instructions when the embedding is broken into
+multiple maven artifacts.
+
+### Step 2: Initialize Flutter
+
+Flutter requires that it be initialized before attempting to display any Flutter
+content. This can be done in one of three ways: 
+
+ * Use `FlutterApplication` as your app's `Application` class.
+
+ * Extend `FlutterApplication` with your own `Application` class.
+
+ * Manually initialize Flutter.
+
+#### Option 1: Use FlutterApplication
+
+To use `FlutterApplication` directly, simply declare it as the `Application`
+class in the `AndroidManifest.xml` file:
+
+```xml
+<application
+  android:name="io.flutter.app.FlutterApplication"
+  // other properties omitted
+  >
+  <!-- content omitted -->
+</application>
+```
+
+Now, as soon as your application is created, Flutter initializes.
+
+#### Option 2: Extend FlutterApplication
+
+Extending `FlutterApplication` automatically initializes Flutter when
+Android creates your application. First, create an `Application` class that
+extends `FlutterApplication`:
+
+```java
+public class MyApplication extends FlutterApplication {
+  // no implementation is necessary.
+}
+```
+
+Second, register your custom `Application` subclass in the
+`AndroidManifest.xml` file:
+
+```xml
+<application
+  android:name=".MyApplication"
+  // other properties omitted
+  >
+  <!-- content omitted -->
+</application>
+```
+
+Replace `MyApplication` with your `Application` subclass name.
+
+#### Option 3: Manually initialize Flutter
+
+If you'd like to control precisely where and when Flutter is initialized, you
+can initialize Flutter manually. This must occur before you attempt to display
+any Flutter content, or execute any Dart code. Initialization only needs to
+happen once within the application process - it does not need to be repeated
+for every Flutter experience.
+
+To manually initialize Flutter, execute the following code at the desired
+location:
+
+```java
+FlutterMain.startInitialization(appContext);
+FlutterMain.ensureInitializationComplete(appContext, null);
```

Comment:
Even if we used FlutterApplication, they still have to call this at some point right? (looking at the source, FlutterApplication only calls FlutterMain.startInitialization)

Reply:
Yeah

Reply:
That part wasn't very clean in https://github.com/flutter/flutter/wiki/Experimental:-Add-Flutter-Activity. Let's call out explicitly that you need to do this for option 1/2/3.

Reply:
This area was rebased away. We should file a new issue if we think a change to this file is required.

Reply:
Ah, I was wondering why both pages were appearing in this PR. Filed https://github.com/flutter/flutter/issues/37195